From 8b5647118663ead8ee839449896d8e305b200b96 Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Thu, 6 Jan 2022 17:09:27 +0100
Subject: [PATCH 58/64] Revert "bfq: Avoid false marking of bic as stably
 merged"

This reverts commit cc144bc45e2814f4ca7b90ad42ad3d70fe333b38.

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 block/bfq-iosched.c | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/block/bfq-iosched.c b/block/bfq-iosched.c
index 1ab9404..332415c 100644
--- a/block/bfq-iosched.c
+++ b/block/bfq-iosched.c
@@ -2902,12 +2902,9 @@ bfq_setup_cooperator(struct bfq_data *bfqd, struct bfq_queue *bfqq,
 				struct bfq_queue *new_bfqq =
 					bfq_setup_merge(bfqq, stable_merge_bfqq);
 
-				if (new_bfqq) {
-					bic->stably_merged = true;
-					if (new_bfqq->bic)
-						new_bfqq->bic->stably_merged =
-									true;
-				}
+				bic->stably_merged = true;
+				if (new_bfqq && new_bfqq->bic)
+					new_bfqq->bic->stably_merged = true;
 				return new_bfqq;
 			} else
 				return NULL;
-- 
2.34.1.75.gabe6bb3905

